connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -80,7 +99,26 @@ class UCSingleCatalog extends TableCatalog with SupportsNamespaces with Logging
       // user-specified but system-generated, which is exactly the case here.
       newProps.put(TableCatalog.PROP_IS_MANAGED_LOCATION, "true")
       delegate.createTable(ident, columns, partitions, newProps)
+    } else if (hasLocationClause) {
+      val location = properties.get(TableCatalog.PROP_LOCATION)
+      assert(location != null)
+      val cred = temporaryCredentialsApi.generateTemporaryPathCredentials(
+        new GenerateTemporaryPathCredential().url(location).operation(PathOperation.PATH_CREATE_TABLE))
+      val newProps = new util.HashMap[String, String]
+      newProps.putAll(properties)
+      val credentialProps = UCSingleCatalog.generateCredentialProps(
+        CatalogUtils.stringToURI(location).getScheme, cred)
+      newProps.putAll(credentialProps.asJava)
+      // TODO: Delta requires the options to be set twice in the properties, with and without the
+      //       `option.` prefix. We should revisit this in Delta.
+      val prefix = TableCatalog.OPTION_PREFIX
+      newProps.putAll(credentialProps.map {
+        case (k, v) => (prefix + k, v)
+      }.asJava)
+      delegate.createTable(ident, columns, partitions, newProps)
     } else {
+      // TODO: for path-based tables, Spark should generate a location property using the qualified
```

Comment:
what does this TODO mean? is there some changed needed here ... or in Spark? 
will create on path based tables invoke `UCSingleCatalog`?

Reply:
In DBR, the path string in the path identifier will be qualified in the DeltaCatalog, then call UC to request credentials. Now we are requesting credentials before entering DeltaCatalog, and the path is not qualified yet. We should let Spark to qualify the path string in the path identifier, similar to how Spark handle the normal table location.